src/js/helpers/generalHelpers.ts

```diff
@@ -18,30 +18,35 @@ export const getFieldValueFromKeyString = <T extends Value>(
 };
 
 const setDeepValue = <T extends Value>(
-    value: Value, fieldNames: string[], currentIndex: number, oldValues: Value
+    value: Value, fieldNames: string[], currentIndex: number, oldValues: Value,
 ): T => {
-    if (typeof oldValues !== 'object' && !Array.isArray(oldValues)) {
-        throw new Error('Could not find value for ' + fieldNames.join('.'));
+    let ret;
+    if (typeof oldValues !== 'object') {
+        ret = Number.isNaN(Number(fieldNames[currentIndex])) ? {} : [];
```

Comment:
It took me a while to understand the if clause of this ternary operator. It is basically just checking if at the current index there is a number in the fieldNames, which means the value at this point in the shape is an array, correct? I wonder if there's an easier to understand way of checking that, but I'm not sure...

Reply:
Yes, that's exactly what it does. I don't really know how else I could do it tbh